wp-content/themes/pub/wporg-learn-2024/functions.php

```diff
@@ -60,6 +61,27 @@ function modify_single_template( $templates ) {
 	return $templates;
 }
 
+/**
+ * Modify the search template hierarchy to use search-front-page templates.
+ *
+ * @param array $templates Array of template files.
+ * @return array
+ */
+function modify_search_template( $templates ) {
+	// Should not change the search result template of course, lesson, and learning-pathway.
+	// Currently, they each use their specific templates: archive-course, archive-lesson, and taxonomy-learning-pathway,
+	// which have their own dedicated UI and filters.
+	if (
+		is_search() &&
+		! ( is_post_type_archive( 'course' ) || is_post_type_archive( 'lesson' ) ) &&
+		! is_tax( 'learning-pathway' )
+		) {
+			array_unshift( $templates, 'search-front-page' );
```

Comment:
Is there a better name for [this template](https://github.com/WordPress/Learn/blob/c66d3e6da5704ac016c8a0de4682995aa0506ce5/wp-content/themes/pub/wporg-learn-2024/templates/search-front-page.html)?

Reply:
What about `search-all`?

Reply:
Thanks! 935d259